src/inet/InetLayer.cpp

```diff
@@ -975,18 +975,13 @@ void InetLayer::HandleTCPInactivityTimer(chip::System::Layer * aSystemLayer, voi
     InetLayer & lInetLayer = *reinterpret_cast<InetLayer *>(aAppState);
     bool lTimerRequired    = lInetLayer.IsIdleTimerRunning();
 
-    for (size_t i = 0; i < INET_CONFIG_NUM_TCP_ENDPOINTS; i++)
-    {
-        TCPEndPoint * lEndPoint = TCPEndPoint::sPool.Get(*aSystemLayer, i);
-
-        if (lEndPoint == nullptr)
-            continue;
+    TCPEndPoint::sPool.ForEachActiveObject([&](TCPEndPoint * lEndPoint) {
         if (!lEndPoint->IsCreatedByInetLayer(lInetLayer))
-            continue;
+            return true;
         if (!lEndPoint->IsConnected())
-            continue;
+            return true;
         if (lEndPoint->mIdleTimeout == 0)
-            continue;
+            return true;
```

Comment:
Why the change to `return true` from `continue` ? This changes the semantics of the loop. Not sure of the original semantics (sure could use some comments), but is it intended?

Reply:
Return true if you want to skip to the next iteration in the ForEachActiveObject loop, return false if you want to jump out of ForEachActiveObject loop.